imagepicker/src/main/java/com/esafirm/imagepicker/features/ImagePickerFragment.java

```diff
@@ -534,8 +536,53 @@ public void showCapturedImage() {
     @Override
     public void showFetchCompleted(List<Image> images, List<Folder> folders) {
         ImagePickerConfig config = getImagePickerConfig();
+
         if (config != null && config.isFolderMode()) {
-            setFolderAdapter(folders);
+            List<Folder> sortedFolder = new ArrayList<>();
+            ArrayList<Image> allImages = new ArrayList<>(images);
+
+            Folder allMediaFolder = new Folder(Constants.ALL_MEDIA_FOLDER);
+            allMediaFolder.setImages(allImages);
+            folders.add(allMediaFolder);
+
+            for (Folder folder : folders) {
+                if (folder.getFolderName().equals(Constants.CAMERA_FOLDER)) {
+                    sortedFolder.add(folder);
+                } else {
+                    Timber.d("Folder Camera not found");
+                }
+            }
+
+            for (Folder folder : folders) {
+                if (folder.getFolderName().equals(Constants.ALL_MEDIA_FOLDER)) {
+                    sortedFolder.add(folder);
+                } else {
+                    Timber.d("Folder All Media not found");
+                }
+            }
+
+            for (Folder folder : folders) {
+                if (folder.getFolderName().equals(Constants.HI_APP_IMAGES_FOLDER)) {
+                    sortedFolder.add(folder);
+                } else {
+                    Timber.d("Folder HiAppImages not found");
+                }
+            }
+            for (Folder folder : folders) {
+                if (folder.getFolderName().equals(Constants.HI_APP_WALLPAPER_FOLDER)) {
+                    sortedFolder.add(folder);
+                } else {
+                    Timber.d("Folder HiAppWallpaper not found");
+                }
+            }
+            for (Folder folder : folders) {
+                if (!folder.getFolderName().equals(Constants.HI_APP_WALLPAPER_FOLDER) &&
+                        !folder.getFolderName().equals(Constants.HI_APP_IMAGES_FOLDER) &&
+                        !folder.getFolderName().equals(Constants.CAMERA_FOLDER)) {
+                    sortedFolder.add(folder);
+                }
```

Comment:
nanya mas @syseno, ini perulangannya buat sorting harus satu per satu yaa? kalo misal dibikin if-elseif di dalem 1 for aja gimana?

Reply:
Gak bisa mbak, soalnya kalau barengan kita gak tau nanti yg mana dluan folder yg masuk pas sorting...